<<<<<<< HEAD
---
description: 'Enforce valid definition of `new` and `constructor`.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-misused-new** for documentation.
=======
# `no-misused-new`

Enforce valid definition of `new` and `constructor`.
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

Warns on apparent attempts to define constructors for interfaces or `new` for classes.

## Rule Details

Examples of code for this rule:

<!--tabs-->

### ‚ùå Incorrect

```ts
class C {
  new(): C;
}

interface I {
  new (): I;
  constructor(): void;
}
```

### ‚úÖ Correct

```ts
class C {
  constructor() {}
}
interface I {
  new (): C;
}
```
<<<<<<< HEAD
=======

## Options

```jsonc
// .eslintrc.json
{
  "rules": {
    "@typescript-eslint/no-misused-new": "error"
  }
}
```

This rule is not configurable.

## Related To

- TSLint: [no-misused-new](https://palantir.github.io/tslint/rules/no-misused-new/)

## Attributes

- [x] ‚úÖ Recommended
- [ ] üîß Fixable
- [ ] üí≠ Requires type information
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.
