<<<<<<< HEAD
---
description: 'Disallow invocation of `require()`.'
---

> ğŸ›‘ This file is source code, not the primary documentation location! ğŸ›‘
>
> See **https://typescript-eslint.io/rules/no-require-imports** for documentation.
=======
# `no-require-imports`

Disallows invocation of `require()`.
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

Prefer the newer ES6-style imports over `require()`.

## Rule Details

Examples of code for this rule:

<!--tabs-->

### âŒ Incorrect

```ts
var lib = require('lib');
let lib2 = require('lib2');
var lib5 = require('lib5'),
  lib6 = require('lib6');
import lib8 = require('lib8');
```

### âœ… Correct

```ts
import { l } from 'lib';
var lib3 = load('not_an_import');
var lib4 = lib2.subImport;
var lib7 = 700;
import lib9 = lib2.anotherSubImport;
import lib10 from 'lib10';
```

## When Not To Use It
<<<<<<< HEAD
=======

If you don't care about TypeScript module syntax, then you will not need this rule.

## Related To

- TSLint: [no-require-imports](https://palantir.github.io/tslint/rules/no-require-imports/)
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

## Attributes

- [ ] âœ… Recommended
- [ ] ğŸ”§ Fixable
- [ ] ğŸ’­ Requires type information
