<<<<<<< HEAD
---
description: 'Disallow unused variables.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-unused-vars** for documentation.
=======
# `no-unused-vars`

Disallow unused variables.
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

## Rule Details

This rule extends the base [`eslint/no-unused-vars`](https://eslint.org/docs/rules/no-unused-vars) rule.
It adds support for TypeScript features, such as types.
<<<<<<< HEAD
=======

## How to Use

```jsonc
{
  // note you must disable the base rule as it can report incorrect errors
  "no-unused-vars": "off",
  "@typescript-eslint/no-unused-vars": ["error"]
}
```

## Options

See [`eslint/no-unused-vars` options](https://eslint.org/docs/rules/no-unused-vars#options).

<sup>

Taken with ‚ù§Ô∏è [from ESLint core](https://github.com/eslint/eslint/blob/main/docs/rules/no-unused-vars.md)

</sup>

## Attributes

- [x] ‚úÖ Recommended
- [ ] üîß Fixable
- [ ] üí≠ Requires type information
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.
