<<<<<<< HEAD
---
description: 'Require both operands of addition to have type `number` or `string`.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/restrict-plus-operands** for documentation.
=======
# `restrict-plus-operands`

When adding two variables, operands must both be of type number or of type string.
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

## Rule Details

Examples of code for this rule:

<!--tabs-->

### ‚ùå Incorrect

```ts
var foo = '5.5' + 5;
var foo = 1n + 1;
```

### ‚úÖ Correct

```ts
var foo = parseInt('5.5', 10) + 10;
var foo = 1n + 1n;
```

## Options

### `checkCompoundAssignments`

Examples of code for this rule with `{ checkCompoundAssignments: true }`:

<!--tabs-->

#### ‚ùå Incorrect

```ts
/*eslint @typescript-eslint/restrict-plus-operands: ["error", { "checkCompoundAssignments": true }]*/

let foo: string | undefined;
foo += 'some data';

let bar: string = '';
bar += 0;
```

#### ‚úÖ Correct

```ts
/*eslint @typescript-eslint/restrict-plus-operands: ["error", { "checkCompoundAssignments": true }]*/

let foo: number = 0;
foo += 1;

let bar = '';
bar += 'test';
```

### `allowAny`

Examples of code for this rule with `{ allowAny: true }`:

<!--tabs-->

#### ‚ùå Incorrect

```ts
var fn = (a: any, b: boolean) => a + b;
var fn = (a: any, b: []) => a + b;
var fn = (a: any, b: {}) => a + b;
```

#### ‚úÖ Correct

```ts
var fn = (a: any, b: any) => a + b;
var fn = (a: any, b: string) => a + b;
var fn = (a: any, b: bigint) => a + b;
var fn = (a: any, b: number) => a + b;
```

## How to Use
<<<<<<< HEAD
=======

```json
{
  "@typescript-eslint/restrict-plus-operands": "error"
}
```

## Related To

- TSLint: [restrict-plus-operands](https://palantir.github.io/tslint/rules/restrict-plus-operands/)

## Attributes

- [x] ‚úÖ Recommended
- [ ] üîß Fixable
- [x] üí≠ Requires type information
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.
