<<<<<<< HEAD
---
description: 'Disallow two overloads that could be unified into one with a union or an optional/rest parameter.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/unified-signatures** for documentation.
=======
# `unified-signatures`

Warns for any two overloads that could be unified into one by using a union or an optional/rest parameter.
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

## Rule Details

This rule aims to keep the source code as maintainable as possible by reducing the amount of overloads.

<<<<<<< HEAD
Examples of code for this rule with the default options:
=======
Examples of code for this rule:
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

<!--tabs-->

### ‚ùå Incorrect

```ts
function x(x: number): void;
function x(x: string): void;
```

```ts
function y(): void;
function y(...x: number[]): void;
```

### ‚úÖ Correct

```ts
function x(x: number | string): void;
```

```ts
function y(...x: number[]): void;
```

<<<<<<< HEAD
## Options

### `ignoreDifferentlyNamedParameters`

Examples of code for this rule with `ignoreDifferentlyNamedParameters`:
=======
## Related To
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.

- TSLint: [`unified-signatures`](https://palantir.github.io/tslint/rules/unified-signatures/)

## Attributes

<<<<<<< HEAD
```ts
function f(a: number): void;
function f(a: string): void;
```

```ts
function f(...a: number[]): void;
function f(...b: string[]): void;
```

### ‚úÖ Correct

```ts
function f(a: number): void;
function f(b: string): void;
```

```ts
function f(...a: number[]): void;
function f(...a: string[]): void;
```

## Options
=======
- [ ] ‚úÖ Recommended
- [ ] üîß Fixable
- [ ] üí≠ Requires type information
>>>>>>> parent of 8c1a3d0d6... GitHub Pages fix.
